### PR TITLE
NIFI-5623 Upgraded OkHttp to 4.9.1 and updated unit tests

### PR DESCRIPTION
**Pull request by exceptionfactory**
#### Description of PR

NIFI-5623 Upgrades OkHttp3 to version 4.9.1 and updates several unit tests to work with the new version.

- Created `okhttp.version` property in root Maven configuration to reflect version used in NiFi framework components
- Refactored unit test for `HttpNotificationService` to remove unnecessary test methods and streamline testing approach
- `HttpNotificationServiceTest` includes URL property creation to work on all platforms following OkHttp version updates for OkHttp `MockWebServer`, and also removes the need for binary JKS files in `nifi-bootstrap` test resources
- Preserved OkHttp version override in `nifi-toolkit-api` to maintain support for generated classes

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [X] Have you written or updated unit tests to verify your changes?
- [X] Have you verified that the full build is successful on JDK 8?
- [X] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/pom.xml [hunk 1]
```diff
@@ -74,7 +74,7 @@
         <dependency>
             <groupId>com.squareup.okhttp3</groupId>
             <artifactId>okhttp</artifactId>
-            <version>3.14.4</version>
+            <version>${okhttp.version}</version>
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/pom.xml:77]: Any reason to use the "externalized" version for NiFi framework here but not for the other bundles? The OAuth2 bundle is not part of the framework pieces I believe.

**Review by pvillard31 — COMMENTED** [review]
LGTM, just left a comment for a specific version in the OAuth2 bundle.

File: nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/pom.xml (hunk 1 above)
Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/pom.xml:77]: Thanks for the feedback @pvillard31, that's a good point, I didn't intend to use the framework version in this bundle, will make the change here to use the explicit version.

**Review by pvillard31 — APPROVED** [review]

**Comment by pvillard31** [comment]
Merged to main, thanks @exceptionfactory

**Pull request closed without merging**